dashboard/modules/stats_collector/tests/test_stats_collector.py

```diff
@@ -204,6 +207,50 @@ def _check_nodes():
 
     wait_for_condition(_check_nodes, timeout=10)
 
+@pytest.mark.parametrize(
+    "ray_start_cluster_head", [{
+        "include_dashboard": True
+    }], indirect=True)
+def test_multi_node_churn(enable_test_module, disable_aiohttp_cache,
+                          ray_start_cluster_head):
+    cluster: Cluster = ray_start_cluster_head
+    assert (wait_until_server_available(cluster.webui_url) is True)
+    webui_url = format_web_url(cluster.webui_url)
+
+    def cluster_chaos_monkey():
+        worker_nodes = []
+        while True:
+            time.sleep(5)
+            if len(worker_nodes) < 2:
+                worker_nodes.append(cluster.add_node())
+                continue
+            should_add_node = random.randint(0, 1)
+            if should_add_node:
+                worker_nodes.append(cluster.add_node())
+            else:
+                node_index = random.randrange(0, len(worker_nodes))
+                node_to_remove = worker_nodes.pop(node_index)
+                cluster.remove_node()
+
+    def get_index():
+        resp = requests.get(webui_url)
+        resp.raise_for_status()
+
+    def get_nodes():
+        resp = requests.get(webui_url + "/nodes?view=summary")
+        resp.raise_for_status()
+        summary = resp.json()
+        assert summary["result"] is True, summary["msg"]
+        assert summary["data"]["summary"]
+
+    t = threading.Thread(target=cluster_chaos_monkey, daemon=True)
+    t.start()
```

Comment:
Could consider using a ray actor for this as well

Reply:
@edoakes I like that idea. What's the best way to force my actor to live on the head node? Otherwise I'm concerned that it'll get destroyed by a node getting taken down (although I guess it would then be restarted on a node that's still up).

Reply:
@mfitton you can create a custom resource on the head node and target that resource with `Actor.options(resources=xxx)`.